lib/LLVMIRCodeGen/GlowJIT.cpp

```diff
@@ -72,36 +72,17 @@ class NotifyLoadedFunctor {
       : dbgRegistrationListener_(
             llvm::JITEventListener::createGDBRegistrationListener()) {}
 
-#if LLVM_VERSION_MAJOR <= 6
-  void operator()(llvm::orc::RTDyldObjectLinkingLayerBase::ObjHandleT,
-                  const llvm::orc::RTDyldObjectLinkingLayerBase::ObjectPtr &obj,
-                  const llvm::RuntimeDyld::LoadedObjectInfo &objInfo) {
-    auto loadedObj = obj->getBinary();
-    // Inform the debugger about the loaded object file. This should allow for
-    // more complete stack traces under debugger. And even it should even enable
-    // the stepping functionality on platforms supporting it.
-    dbgRegistrationListener_->NotifyObjectEmitted(*loadedObj, objInfo);
-    // Dump symbol information for the JITed symbols.
-    dumpSymbolInfo(*loadedObj, objInfo);
-  }
-#else
   void operator()(llvm::orc::VModuleKey key,
                   const llvm::object::ObjectFile &obj,
                   const llvm::RuntimeDyld::LoadedObjectInfo &objInfo) {
     auto &loadedObj = obj;
     // Inform the debugger about the loaded object file. This should allow for
     // more complete stack traces under debugger. And even it should even enable
     // the stepping functionality on platforms supporting it.
-#if LLVM_VERSION_MAJOR < 8 || FACEBOOK_INTERNAL
     dbgRegistrationListener_->NotifyObjectEmitted(loadedObj, objInfo);
-#else
```

Comment:
Why did you remove this snippet? This was introduced for LLVM >=8, IIRC.

Reply:
This passed in the internal version, therefore I wrongfully assumed it passes for llvm8. It doesn't built locally. reverting all of these patches.